x-pack/plugins/siem/server/graphql/kpi_hosts/kpi_hosts.mock.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from '../../utils/logger';
+import { SiemContext } from '../index';
+import { KpiHostsData } from '../types';
+
+export const mockKpiHostsData: { KpiHosts: KpiHostsData } = {
```

Comment:
The data here looks strange, have I done anything wrong?

Reply:
I don't think doc_count should be a float or negative float. I'm pretty sure it's always a positive integer.

Here is an output I ran just now from using `console.log` statements.

```ts
raw return:
{
  "took": 2641,
  "responses": [
    {
      "took": 2641,
      "timed_out": false,
      "_shards": {
        "total": 26,
        "successful": 26,
        "skipped": 0,
        "failed": 0
      },
      "hits": {
        "max_score": null,
        "hits": []
      },
      "aggregations": {
        "unique_destination_ips_histogram": {
          "buckets": [
            {
              "key_as_string": "2019-05-07T19:00:00.000Z",
              "key": 1557255600000,
              "doc_count": 2076898,
              "count": {
                "value": 1514
              }
            },
            {
              "key_as_string": "2019-05-08T07:00:00.000Z",
              "key": 1557298800000,
              "doc_count": 3008198,
              "count": {
                "value": 1588
              }
            },
            {
              "key_as_string": "2019-05-08T19:00:00.000Z",
              "key": 1557342000000,
              "doc_count": 86515,
              "count": {
                "value": 232
              }
            }
          ],
          "interval": "12h"
        },
        "unique_source_ips": {
          "value": 1738
        },
        "hosts": {
          "value": 1234
        },
        "unique_source_ips_histogram": {
          "buckets": [
            {
              "key_as_string": "2019-05-07T19:00:00.000Z",
              "key": 1557255600000,
              "doc_count": 2076898,
              "count": {
                "value": 1081
              }
            },
            {
              "key_as_string": "2019-05-08T07:00:00.000Z",
              "key": 1557298800000,
              "doc_count": 3008198,
              "count": {
                "value": 1020
              }
            },
            {
              "key_as_string": "2019-05-08T19:00:00.000Z",
              "key": 1557342000000,
              "doc_count": 86515,
              "count": {
                "value": 191
              }
            }
          ],
          "interval": "12h"
        },
        "hosts_histogram": {
          "buckets": [
            {
              "key_as_string": "2019-05-07T19:00:00.000Z",
              "key": 1557255600000,
              "doc_count": 2076898,
              "count": {
                "value": 623
              }
            },
            {
              "key_as_string": "2019-05-08T07:00:00.000Z",
              "key": 1557298800000,
              "doc_count": 3008198,
              "count": {
                "value": 586
              }
            },
            {
              "key_as_string": "2019-05-08T19:00:00.000Z",
              "key": 1557342000000,
              "doc_count": 86515,
              "count": {
                "value": 53
              }
            }
          ],
          "interval": "12h"
        },
        "unique_destination_ips": {
          "value": 2380
        }
      },
      "status": 200
    },
    {
      "took": 833,
      "timed_out": false,
      "_shards": {
        "total": 26,
        "successful": 26,
        "skipped": 0,
        "failed": 0
      },
      "hits": {
        "max_score": null,
        "hits": []
      },
      "aggregations": {
        "authentication_success": {
          "doc_count": 23
        },
        "authentication_failure": {
          "doc_count": 18927
        },
        "authentication_failure_histogram": {
          "buckets": [
            {
              "key_as_string": "2019-05-07T19:00:00.000Z",
              "key": 1557255600000,
              "doc_count": 14824,
              "count": {
                "doc_count": 14822
              }
            },
            {
              "key_as_string": "2019-05-08T07:00:00.000Z",
              "key": 1557298800000,
              "doc_count": 4007,
              "count": {
                "doc_count": 3989
              }
            },
            {
              "key_as_string": "2019-05-08T19:00:00.000Z",
              "key": 1557342000000,
              "doc_count": 119,
              "count": {
                "doc_count": 116
              }
            }
          ],
          "interval": "12h"
        },
        "authentication_success_histogram": {
          "buckets": [
            {
              "key_as_string": "2019-05-07T19:00:00.000Z",
              "key": 1557255600000,
              "doc_count": 14824,
              "count": {
                "doc_count": 2
              }
            },
            {
              "key_as_string": "2019-05-08T07:00:00.000Z",
              "key": 1557298800000,
              "doc_count": 4007,
              "count": {
                "doc_count": 18
              }
            },
            {
              "key_as_string": "2019-05-08T19:00:00.000Z",
              "key": 1557342000000,
              "doc_count": 119,
              "count": {
                "doc_count": 3
              }
            }
          ],
          "interval": "12h"
        }
      },
      "status": 200
    }
  ]
}
```

```ts
{
  "hosts": 1234,
  "hostsHistogram": [
    {
      "key_as_string": "2019-05-07T19:00:00.000Z",
      "key": 1557255600000,
      "doc_count": 2076898,
      "count": {
        "value": 623
      }
    },
    {
      "key_as_string": "2019-05-08T07:00:00.000Z",
      "key": 1557298800000,
      "doc_count": 3008198,
      "count": {
        "value": 586
      }
    },
    {
      "key_as_string": "2019-05-08T19:00:00.000Z",
      "key": 1557342000000,
      "doc_count": 86515,
      "count": {
        "value": 53
      }
    }
  ],
  "authSuccess": 23,
  "authSuccessHistogram": [
    {
      "key_as_string": "2019-05-07T19:00:00.000Z",
      "key": 1557255600000,
      "doc_count": 14824,
      "count": {
        "doc_count": 2
      }
    },
    {
      "key_as_string": "2019-05-08T07:00:00.000Z",
      "key": 1557298800000,
      "doc_count": 4007,
      "count": {
        "doc_count": 18
      }
    },
    {
      "key_as_string": "2019-05-08T19:00:00.000Z",
      "key": 1557342000000,
      "doc_count": 119,
      "count": {
        "doc_count": 3
      }
    }
  ],
  "authFailure": 18927,
  "authFailureHistogram": [
    {
      "key_as_string": "2019-05-07T19:00:00.000Z",
      "key": 1557255600000,
      "doc_count": 14824,
      "count": {
        "doc_count": 14822
      }
    },
    {
      "key_as_string": "2019-05-08T07:00:00.000Z",
      "key": 1557298800000,
      "doc_count": 4007,
      "count": {
        "doc_count": 3989
      }
    },
    {
      "key_as_string": "2019-05-08T19:00:00.000Z",
      "key": 1557342000000,
      "doc_count": 119,
      "count": {
        "doc_count": 116
      }
    }
  ],
  "uniqueSourceIps": 1738,
  "uniqueSourceIpsHistogram": [
    {
      "key_as_string": "2019-05-07T19:00:00.000Z",
      "key": 1557255600000,
      "doc_count": 2076898,
      "count": {
        "value": 1081
      }
    },
    {
      "key_as_string": "2019-05-08T07:00:00.000Z",
      "key": 1557298800000,
      "doc_count": 3008198,
      "count": {
        "value": 1020
      }
    },
    {
      "key_as_string": "2019-05-08T19:00:00.000Z",
      "key": 1557342000000,
      "doc_count": 86515,
      "count": {
        "value": 191
      }
    }
  ],
  "uniqueDestinationIps": 2380,
  "uniqueDestinationIpsHistogram": [
    {
      "key_as_string": "2019-05-07T19:00:00.000Z",
      "key": 1557255600000,
      "doc_count": 2076898,
      "count": {
        "value": 1514
      }
    },
    {
      "key_as_string": "2019-05-08T07:00:00.000Z",
      "key": 1557298800000,
      "doc_count": 3008198,
      "count": {
        "value": 1588
      }
    },
    {
      "key_as_string": "2019-05-08T19:00:00.000Z",
      "key": 1557342000000,
      "doc_count": 86515,
      "count": {
        "value": 232
      }
    }
  ]
}
```

Reply:
Thanks for logging the result, so probably my test didn't even run correctly, as it should have failed?